src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1435,7 +1435,7 @@ public DynamicJsonArray GetTimeSeriesNamesForDocument(DocumentsOperationContext
                 while (table.SeekOneBackwardByPrimaryKeyPrefix(documentKeyPrefix, slice, out reader))
                 {
                     name = UpdateSlice(ref reader, documentKeyPrefix.Size, ref slice);
-                    list.Add(name);
+                    list.Insert(0, name);
```

Comment:
That is going to be costly. Better to add, then reverse in the end

Reply:
done